lib/src/local_working_copy.rs

```diff
@@ -1062,8 +1082,17 @@ impl TreeState {
         if clean {
             Ok(None)
         } else {
-            let new_file_type = new_file_state.file_type.clone();
             let current_tree_values = current_tree.path_value(repo_path);
+            #[cfg(windows)]
+            let new_file_type = {
+                let mut new_file_type = new_file_state.file_type.clone();
+                if matches!(current_tree_values.as_normal(), Some(TreeValue::Symlink(_))) {
```

Comment:
nit: might be better to `match new_file_type` too. If the disk file type is `GitSubmodule` (or other unknown types to be added in future), it shouldn't be translated to `Symlink`.

Reply:
Good point.